### DISCUSSION TITLE
Serverless function timed out after 10.01 seconds, yet local and build completes in less than 2 seconds

### DISCUSSION THREAD

**Discussion by alex-r89** [vercel/next.js#28239] [Help]
Hi there,

I did see [this issue](https://github.com/vercel/next.js/discussions/16957) , which talks about the 10 second timeout being hit, but I couldn't ascertain much information from it related to the issue im facing.

I have a route which is continually hitting the error `Task timed out after 10.01 seconds` in production. This route uses `getStaticProps` with the `revalidate` prop (set as 86400; once per day) to get page information at build time by calling an external API.

To work out why I was hitting this timeout, I crudely added a console log (`new Date().getTime() / 1000`) at the start and at the end of `getStaticProps`. This shows me the current time in seconds and means I can compare the first value to the last value to calculate how long the entire `getStaticProps` took.

This issue I have however is that locally, the time between the first `getTime` and the last `getTime` is around 1-2 seconds. This completely confused me because I do not see how, locally, `getStaticProps`can be so fast yet on vercel it can take 10x longer than this and cause the time out.

**To further add to the confusion**, I pushed the code to vercel with the console logs in `getStaticProps` and sure enough just like with local, the time between the first console log and the last is around 1-2 seconds in the vercel build logs 🤯.

Could anyone advise on what could be happening, or what more I could do? If the times I was seeing locally and in build were close to 10 seconds I would conclude that it must be a slow api. But the fact that this is not only close to 10x quicker locally, but also almost 10x quicker in the vercel build logs, leaves me in the dark about why this takes so long when it revalidates.

Thanks

**Comment by HumidBrains** [comment]
I got the same problem. For me this happens sometimes on navigation between pages and always when adding a product to cart on the e-commerce I'm building. And it's been impossible to find any info about this so far.
